app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -0,0 +1,63 @@
+module ManageIQ::Providers
+  class Hawkular::MiddlewareManager < ManageIQ::Providers::MiddlewareManager
+    require_nested :RefreshParser
+    require_nested :RefreshWorker
+    require_nested :Refresher
+
+    include AuthenticationMixin
+
+    DEFAULT_PORT = 80
+    default_value_for :port, DEFAULT_PORT
+    has_many :middleware_servers, :foreign_key => :ems_id, :dependent => :destroy
+    has_many :middleware_deployments, :foreign_key => :ems_id, :dependent => :destroy
+
+    def verify_credentials(auth_type = nil, _options = {})
+      auth_type || 'default'
+      connect(_options).list_feeds
+    end
+
+    def self.raw_connect(hostname, port, username, password)
+      require 'hawkular_all'
+      url = URI::HTTP.build(:host => hostname, :port => port.to_i, :path => '/hawkular/inventory').to_s
+      ::Hawkular::Inventory::InventoryClient.new(url, :username => username, :password => password)
+    end
+
+    def connect(_options = {})
+      require 'hawkular_all'
+      $log.debug("Connecting to Hawkular on #{hostname}:#{port}")
+      self.class.raw_connect(hostname,
+                             port,
+                             authentication_userid('default'),
+                             authentication_password('default'))
+    end
+
+    def feeds
+      with_provider_connection(&:list_feeds)
+    end
+
+    def eaps(feed)
+      with_provider_connection do |connection|
+        connection.list_resources_for_type(feed, 'WildFly Server', true)
+      end
+    end
+
+    def child_resources(eap_parent)
+      with_provider_connection do |connection|
+        connection.list_child_resources(eap_parent)
+      end
+    end
```

Comment:
This is much better :+1:.  I'm still thinking this should be extracted into a helper class that wraps the native client.  Maybe not for this PR, but let's discuss.